R/sdm-rf.R

```diff
@@ -10,31 +11,34 @@ one_rf_tree = function(bbs, vars, sp_id, iter, use_obs_model, x_richness,
     mutate(present = factor(ifelse(is.na(abundance), 0, 1)))
   
   my_formula = as.formula(paste("present ~", 
-                                paste(vars, collapse = "+")))
-   
+                                paste(vars, collapse = " + ")))
+  
   rf = randomForest(
     my_formula, 
     filter(d, in_train),
     ntree = 1
   )  
   
-  test = filter(d, !in_train, iteration == iter)
+  test = make_test_set(d, future, future_observer_effects, settings)
   test$mean = predict(rf, test, type = "prob")[,2]
   test$use_obs_model = use_obs_model
   
   select(test, site_id, year, species_id, mean, richness, use_obs_model, iteration)
 }
 
-rf_predict_species = function(sp_id, bbs, settings, x_richness, use_obs_model){
+rf_predict_species = function(sp_id, bbs, settings, x_richness, use_obs_model,
+                              future, future_observer_effects){
   vars = c(settings$vars, if (use_obs_model) {"observer_effect"})
   iters = sort(unique(x_richness$iteration))
   results = lapply(iters, one_rf_tree,
                    bbs = bbs, sp_id = sp_id, use_obs_model = use_obs_model, 
                    vars = vars,
                    x_richness = x_richness,
-                   last_train_year = settings$last_train_year) %>% 
+                   last_train_year = settings$last_train_year,
+                   future, future_observer_effects,
+                   settings) %>% 
```

Comment:
Since you're using explicit names for all of the other variables I'd probably use them for future, future_observer_effects, and settings, for consistency.